doc/source/whatsnew/v0.23.0.txt

```diff
@@ -308,6 +308,39 @@ The :func:`DataFrame.assign` now accepts dependent keyword arguments for python
 
       df.assign(A=df.A+1, C= lambda df: df.A* -1)
 
+.. _whatsnew_0230.enhancements.str_cat_align:
+
+``Series.str.cat`` has gained the ``join`` kwarg
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Previously, :meth:`Series.str.cat` did not -- in contrast to most of ``pandas`` -- align :class:`Series` on their index before concatenation (see :issue:`18657`).
+The method has now gained a keyword ``join`` to control the manner of alignment. In v.0.23 it will default to None (meaning no alignment), but this default will change
+to ``'left'`` in a future version of pandas.
+
+.. ipython:: python
+
+    s = pd.Series(['a', 'b', 'c', 'd'])
+    t = pd.Series(['b', 'd', 'e', 'c'], index=[1, 3, 4, 2])
+    s.str.cat(t)
+    s.str.cat(t, join='left', na_rep='-')
+
+In particular, ``others`` does not need to be of the same length as the calling ``Series`` (if both have an index and ``join is not None``).
+For more examples, see :ref:`here <text.concatenate>`.
+
+Additionally, ``str.cat`` now allows ``others`` to be a ``DataFrame`` or two-dimensional ``np.ndarray``.
+
+.. ipython:: python
+    
+    u = pd.Series(['b', 'd', 'a', 'c'], index=[1, 3, 0, 2])
+    d = pd.concat([s, u], axis=1)
+    t.str.cat(d.values)
+    s.str.cat(d, join='left', na_rep='-')
+
+Furthermore, any combination of "concatenateable" arguments can be passed in a list-like container (e.g. an iterator).
+    
+For categorical data, it is now possible to call :meth:`Series.str.cat` for ``CategoricalIndex`` as well (previously raised a ``ValueError``).
```

Comment:
are these last 2 points new? is there a separate issue for them?

Reply:
The v.0.22-doc-string of str.cat says "list-like", but actually does not support iterators. I added this.

The error for CategoricalIndex was an oversight in Series.str._validate, and became apparent when writing the test for categorical data in test_strings.py. I also fixed this inconsistency without further prompt.